### PR TITLE
[feature] Add jshint as dependency and invoke task.

### PR DESCRIPTION
**Pull request by abought**
## Purpose
Encourages local testers to run jshint per new Travis requirements in #3107, before pushing to github.

## Summary of changes
- Add jshint to the list of packages that node installs automatically.
- Add a call to jshint (as specified by `.travis.yml`) to the syntax checks called by `inv test --syntax`.
- Consolidated two highly redundant commands (`inv test` and `inv test_all`) into one command with a flag. This will probably inconvenience *someone*...
- Very trivial adjustments for PEP8.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "html-loader": "^0.2.3",
     "imports-loader": "^0.6.3",
     "jquery.cookie": "^1.4.1",
+    "jshint": "^2.8.0",
```
Comment (sloria) [inline package.json]: This should be added in `devDependencies` so that it doesn't get installed on server deployments.

File: tasks.py [hunk 2]
```diff
@@ -783,6 +792,7 @@ def assets(dev=False, watch=False):
     # on prod
     webpack(clean=False, watch=watch, dev=dev)
 
+
```
Comment (sloria) [inline tasks.py]: This is fine, but FWIW, we explicitly ignore the 2-line rule: https://github.com/CenterForOpenScience/osf.io/blob/9ca27b0c5264a97a299fa4fd1c69f3a7914d0365/setup.cfg#L6. It adds unnecessary whitespace without affecting readability at all.

**Comment by sloria** [comment]
Looks good. :octopus:

File: tasks.py (hunk 2 above)
Comment (abought) [inline tasks.py]: Good to know. Found this page which may be helpful for Pycharm users looking to satisfy the built in static analysis feature (which uses the same error codes): http://iambigblind.blogspot.com/2013/02/configuring-pep8py-support-in-pycharm-27.html
